lib/pages/settings_dashboard/settings_profile/settings_profile.dart

```diff
@@ -0,0 +1,264 @@
+            ? ImageSource.camera
+            : ImageSource.gallery,
+        imageQuality: 50,
+      );
+      if (result == null) return;
+      file = MatrixFile(
+        bytes: await result.readAsBytes(),
+        name: result.path,
+      );
+    } else {
+      final result = await FilePicker.platform.pickFiles(
+        type: FileType.image,
+        withData: true,
+      );
+      final pickedFile = result?.files.firstOrNull;
+      if (pickedFile == null) return;
+      file = MatrixFile(
+        bytes: pickedFile.bytes!,
+        name: pickedFile.name,
+      );
+    }
+    final success = await showFutureLoadingDialog(
+      context: context,
+      future: () => matrix.client.setAvatar(file),
+    );
+    if (success.error == null) {
+      _getProfileFromUserId(isUpdated: true);
+    }
+  }
+
+  void setDisplayNameAction() async {
+    if (displayNameFocusNode.hasFocus) {
+      displayNameFocusNode.unfocus();
+    }
+    final matrix = Matrix.of(context);
+    final success = await showFutureLoadingDialog(
+      context: context,
+      future: () => matrix.client.setDisplayName(
+        matrix.client.userID!,
+        displayNameEditingController.text,
+      ),
+    );
+    if (success.error == null) {
+      isEditedProfileNotifier.toggle();
+      _getProfileFromUserId(isUpdated: true);
+    }
+  }
+
+  void _getProfileFromUserId({
+    isUpdated = false,
+  }) async {
+    final profile = await client.getProfileFromUserId(
+      client.userID!,
+      cache: !isUpdated,
+      getFromRooms: false,
+    );
+    Logs().d(
+      'SettingsProfile::_getProfileFromUserId() - avatarUrl: ${profile.avatarUrl} - displayName: ${profile.displayName} - userId: ${profile.userId}',
+    );
+    settingsDashboardManagerController.profileNotifier.value = profile;
+    displayNameEditingController.text = displayName;
+    matrixIdEditingController.text = mxid;
+  }
+
+  void handleTextEditOnChange(SettingsProfileEnum settingsProfileEnum) {
+    switch (settingsProfileEnum) {
+      case SettingsProfileEnum.displayName:
+        _listeningDisplayNameHasChange();
+        break;
+      default:
+        break;
+    }
+  }
+
+  void _listeningDisplayNameHasChange() {
+    isEditedProfileNotifier.value =
+        displayNameEditingController.text != displayName;
+    Logs().d(
+      'SettingsProfile::_listeningDisplayNameHasChange() - ${isEditedProfileNotifier.value}',
+    );
+  }
+
+  void _initProfile() {
+    if (widget.profile == null) {
+      _getProfileFromUserId();
+      return;
+    }
+    settingsDashboardManagerController.profileNotifier.value = widget.profile!;
+    displayNameEditingController.text = displayName;
+    matrixIdEditingController.text = mxid;
+  }
+
+  void copyEventsAction(SettingsProfileEnum settingsProfileEnum) {
+    switch (settingsProfileEnum) {
+      case SettingsProfileEnum.matrixId:
+        Clipboard.setData(ClipboardData(text: mxid));
+        ScaffoldMessenger.of(context).showSnackBar(
```

Comment:
please check my solution for toast (https://github.com/linagora/twake-on-matrix/pull/708). IMO it's better to use, it can be used without any context

Reply:
IMO: Your idea is quite good for me. But we based on Material 3. I think we need to use Snackbars 

https://m3.material.io/components/snackbar/overview

Reply:
I don't mind using snackbars if needed, we just need to make a helper with a static method that we can use without buildcontext @hoangdat wdyt

Reply:
+1 for @imGok